### PR TITLE
Add support for PHP 8.1

### PR DESCRIPTION
**Pull request by internalsystemerror**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | yes

### Description

This adds support to allow installation on PHP 8.1 including any required fixes.

NOTE: This PR is based on the code in PR https://github.com/laminas/laminas-log/pull/28.

### CODE REVIEW

File: phpcs.xml [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
+    <arg value="p"/>
+
+    <!-- set minimal required PHP version (7.3) -->
+    <config name="php_version" value="70400"/>
 
     <!-- Paths to check -->
     <file>src</file>
     <file>test</file>
+
+    <exclude-pattern>src/Int.php</exclude-pattern>
```
Comment (Ocramius) [inline phpcs.xml]: Copied from `laminas-filter`? These classes don't exist here :thinking:

File: phpcs.xml [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+    <exclude-pattern>src/Null.php</exclude-pattern>
```
Comment (Ocramius) [inline phpcs.xml]: Copied from `laminas-filter`? These classes don't exist here :thinking:

**Review by Ocramius — APPROVED** [review]
LGTM :ship:

File: phpcs.xml (hunk 2 above)
Comment (Ocramius) [reply phpcs.xml]: e1cd856

File: phpcs.xml (hunk 1 above)
Comment (Ocramius) [reply phpcs.xml]: e1cd856

File: phpcs.xml (hunk 2 above)
Comment (internalsystemerror) [reply phpcs.xml]: u were correct, ty for the fix

File: phpcs.xml (hunk 1 above)
Comment (internalsystemerror) [reply phpcs.xml]: u were correct, ty for the fix

**Pull request merged**